tests/engine/test_calc_job.py

```diff
@@ -354,62 +354,106 @@ def test_parse_retrieved_folder(self):
+    runner = manager.get_runner()
 
-        for log in expected_logs:
-            assert log in logs
+    process_class = CalculationFactory('arithmetic.add')
+    process = instantiate_process(runner, process_class, **inputs)
+    process.node.set_state(CalcJobState.PARSING)
 
-    def test_parse_not_implemented(self):
-        """Test the scheduler output parsing logic in `CalcJob.parse`.
+    return process
 
-        Here we check explicitly that the parsing does not except even if the scheduler does not implement the method.
-        """
-        process = self.instantiate_process()
-        retrieved = orm.FolderData().store()
-        retrieved.add_incoming(process.node, link_label='retrieved', link_type=LinkType.CREATE)
 
-        process.node.set_attribute('detailed_job_info', {})
+@pytest.mark.usefixtures('clear_database_before_test', 'override_logging')
+def test_parse_insufficient_data(process):
+    """Test the scheduler output parsing logic in `CalcJob.parse`.
+
+    Here we check explicitly that the parsing does not except even if the required information is not available.
+    """
+    retrieved = orm.FolderData().store()
+    retrieved.add_incoming(process.node, link_label='retrieved', link_type=LinkType.CREATE)
+    process.parse()
+
+    filename_stderr = process.node.get_option('scheduler_stderr')
+    filename_stdout = process.node.get_option('scheduler_stdout')
+
+    # The scheduler parsing requires three resources of information, the `detailed_job_info` dictionary which is
+    # stored as an attribute on the calculation job node and the output of the stdout and stderr which are both
+    # stored in the repository. In this test, we haven't created these on purpose. This should not except the
+    # process but should log a warning, so here we check that those expected warnings are attached to the node
+    logs = [log.message for log in orm.Log.objects.get_logs_for(process.node)]
+    expected_logs = [
+        'could not parse scheduler output: the `detailed_job_info` attribute is missing',
+        'could not parse scheduler output: the `{}` file is missing'.format(filename_stderr),
+        'could not parse scheduler output: the `{}` file is missing'.format(filename_stdout)
+    ]
+
+    for log in expected_logs:
+        assert log in logs
+
+
+@pytest.mark.usefixtures('clear_database_before_test', 'override_logging')
+def test_parse_non_zero_retval(process):
+    """Test the scheduler output parsing logic in `CalcJob.parse`.
+
+    This is testing the case where the `detailed_job_info` is incomplete because the call failed. This is checked
+    through the return value that is stored within the attribute dictionary.
+    """
+    retrieved = orm.FolderData().store()
+    retrieved.add_incoming(process.node, link_label='retrieved', link_type=LinkType.CREATE)
+
+    process.node.set_attribute('detailed_job_info', {'retval': 1, 'stderr': 'accounting disabled', 'stdout': ''})
+    process.parse()
```

Comment:
So, isn't the `detailed_job_info` overwritten by the `parse` method of the calculation? Also, I am assuming that you are setting the `retrieved` folder in all these tests because it is necessary for this method.

Reply:
Re the `retrieved` folder, yes this is required by `parse` so that is why I add it. As for the `detailed_job_info`, I am not sure what you mean with overwrites? If you mean in the implementation of `parse_scheduler_output` then yes, but it just overwrites the variable in memory, not the attribute. And it only does this to prevent the `scheduler.parse_output` being called, because anyway that is pointless with a detailed job info that does not contain correct information

Reply:
Mmmm this I didn't fully understand yet though. What do you mean the variable in memory but not the attribute? I don't get the distinction you are making.

Reply:
I was responding to your question:

> So, isn't the detailed_job_info overwritten by the parse method of the calculation?

which I didn't understand. So I guess now we are going in circles of confusion :)

Reply:
Hahaha, I think we can try to clear this out tomorrow. BTW I need to send the email...